extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/controllers.configuration.js

```diff
@@ -491,12 +486,13 @@ angular.module('PaperUI.controllers.configuration',
 	$scope.getThing(true);
 	
 	thingTypeRepository.getOne(function(thingType) {
-		return thingType.UID === thingTypeUID;
+		return thingType.UID === $scope.thingTypeUID;
 	}, function(thingType) {
 		$scope.thingType = thingType;
 		$scope.thingTypeChannels = thingType.channels && thingType.channels.length > 0 ? thingType.channels
 				: thingType.channelGroups;
 		$scope.setHeaderText(thingType.description);
+        $scope.refreshChannels(false);
```

Comment:
Okay, thanks. 
The problem was that the ```getChannels``` function gets called before this http call returns, hence the thingType variable is not set and the ```if (!$scope.thingType || !$scope.thing) {return;}``` statements hits us so that no channels will be displayed in a thing's edit dialog.